junit-jupiter-engine/src/test/java/org/junit/jupiter/api/condition/DisabledIfSystemPropertyConditionTests.java

```diff
@@ -40,13 +40,13 @@ protected ExecutionCondition getExecutionCondition() {
 	}
 
 	@BeforeAll
-	static void setSystemProperty() {
-		DisabledIfSystemPropertyIntegrationTests.setSystemProperty();
```

Comment:
Is it a good idea to call another `@BeforeAll` method like that? (I genuinely don't know, but I find it a bit weird)

Reply:
Maybe there could be an `AbstractSystemPropertyTests` or something like that to handle these shared set-ups? The one in `EnabledIfSystemPropertyIntegrationTests` is also the same

Reply:
> Is it a good idea to call another @BeforeAll method like that? (I genuinely don't know, but I find it a bit weird)

It certainly is a bit strange. I admit.

But I think it's OK for this particular scenario, since the exact same set of system properties are required to make the integration tests and unit tests pass.

Reply:
> The one in EnabledIfSystemPropertyIntegrationTests is also the same.

It's similar but not exactly the same. The two classes uses different prefixes for the keys (i.e., related to the scope of each test class). I did that when these tests were originally introduced in order to avoid possible conflicts (e.g., during potential parallel execution of the test classes). Perhaps that's over-engineering... but it doesn't bother me having it stay like that.

If the rest of the team feels differently about that, we can change it though.